### PR TITLE
User creation request scopes

### PR DESCRIPTION
**Pull request by nyet-ty**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -259,6 +262,22 @@ model UserRole {
   editScheduledDeactivation Boolean @default(false)
   viewScheduledDeactivation Boolean @default(false)
 
+  decideOnUserCreationRequest Boolean @default(false)
+
+  createExistingUserRequest  Boolean @default(false)
+
+  createInternalUserRequest  Boolean @default(false)
+  readManyInternalUserRequest  Boolean @default(false)
```
Comment (tsumo) [inline prisma/schema.prisma:270]: requests

File: prisma/schema.prisma [hunk 2]
```diff
@@ -259,6 +262,22 @@ model UserRole {
...
+  editInternalUserRequest  Boolean @default(false)
+
+  createExternalUserRequest  Boolean @default(false)
+  readManyExternalUserRequest  Boolean @default(false)
```
Comment (tsumo) [inline prisma/schema.prisma:274]: requests

File: prisma/schema.prisma [hunk 3]
```diff
@@ -259,6 +262,22 @@ model UserRole {
...
+  editExternalUserRequest  Boolean @default(false)
+
+  createExternalFromMainUserRequest  Boolean @default(false)
+  readManyExternalFromMainUserRequest  Boolean @default(false)
```
Comment (tsumo) [inline prisma/schema.prisma:278]: same

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/components/PageHeader/PageHeader.tsx [hunk 4]
```diff
@@ -100,6 +100,20 @@ export const PageHeader: React.FC<{ logo?: string; userSettings?: UserSettings }
...
+            createExternalUserRequest: tr('create request to create profile to external employee'),
+            createExternalFromMainUserRequest: tr('createExternalFromMainUserRequest'),
+
+            readManyInternalUserRequest: tr('readManyInternalUserRequest'),
+            readManyExternalUserRequest: tr('readManyExternalUserRequest'),
+            readManyExternalFromMainUserRequest: tr('readManyExternalFromMainUserRequest'),
+
+            editInternalUserRequest: tr('editInternalUserRequest'),
+            editExternalUserRequest: tr('editExternalUserRequest'),
+            editExternalFromMainUserRequest: tr('editExternalFromMainUserRequest'),
```
[the remark this answers is not in the document]
Comment (nyet-ty) [reply src/components/PageHeader/PageHeader.tsx:116]: Thanks!

**Review by LamaEats — APPROVED** [review]

File: src/trpc/router/userCreationRequestRouter.ts [hunk 5]
```diff
@@ -14,9 +15,41 @@ import { processEvent } from '../../utils/analyticsEvent';
 import { dropUnchangedValuesFromEvent } from '../../utils/dropUnchangedValuesFromEvents';
 import { protectedProcedure, router } from '../trpcBackend';
 
+import { tr } from './router.i18n';
+
+const checkRequestTypeAccess = (type: string, forbidden: boolean) => {
```
Comment (tsumo) [inline src/trpc/router/userCreationRequestRouter.ts:20]: `const checkRequestTypeAccess = (types: { type: string, forbidden: boolean }) => {}` would look cleaner

**Review by tsumo — CHANGES_REQUESTED** [review]

**Pull request merged**